src/NuGet.Core/NuGet.Client/ManagedCodeConventions.cs

```diff
@@ -48,6 +48,9 @@ public class ManagedCodeConventions
 
         private RuntimeGraph _runtimeGraph;
 
+        private Dictionary<string, NuGetFramework> _frameworkCache
+            = new Dictionary<string, NuGetFramework>(StringComparer.Ordinal);
```

Comment:
Ordinal gives better performance over OrdinalIgnoreCase for dictionary look ups. The main scenario here is when parsing multiple file paths from the same tfm folder the framework should only be parsed once.